polkadot-protocol/basics/randomness.md

```diff
@@ -1,4 +1,82 @@
+
+Polkadot's VRF is similar to the one used in [**Ouroboros Praos**](https://eprint.iacr.org/2017/573.pdf){target=\_blank}, which secures randomness for block production in systems like [BABE](TODO:update-path){target=\_blank} (Polkadot’s block production mechanism). 
+
+The key difference is that Polkadot's VRF doesn’t rely on a central clock—avoiding the issue of whose clock to trust. Instead, it uses its own past results and slot numbers to simulate time and determine future outcomes.
+
+### How VRF Works
+
+Slots on Polkadot are discrete units of time, each lasting six seconds, and can potentially hold a block. Multiple slots form an epoch, with 2400 slots making up one four-hour epoch.
+
+In each slot, validators execute a **"die roll"** using a **Verifiable Random Function (VRF)**. The VRF uses three inputs:
+
+1. A "secret key", unique to each validator, is used for the die roll
+2. An epoch randomness value, derived from the hash of VRF outputs from blocks two epochs ago (N-2), so past randomness influences the current epoch (N)
+3. The current slot number
+
+This process helps maintain fair randomness across the network.
+
+Here is a graphical representation:
+
+![](/images/polkadot-protocol/basics/blocks-transactions-fees/randomness/slots-epochs.webp)
+
+The VRF produces two outputs: a result (the random number) and a proof (verifying that the number was generated correctly).
+
+The result is checked by the validator against a protocol threshold. If it's below the threshold, the validator becomes a candidate for block production in that slot. 
+
+The validator then attempts to create a block, submitting it along with the `PROOF` and `RESULT`.
+So, VRF can be expressed like:
+
+`(RESULT, PROOF) = VRF(SECRET, EPOCH_RANDOMNESS_VALUE, CURRENT_SLOT_NUMBER)`
+
+Put simply, performing a "VRF roll" generates a random number along with proof that the number was genuinely produced and not arbitrarily chosen.
+
+
+After executing the VRF, the RESULT is compared to a protocol-defined THRESHOLD. If the RESULT is below the THRESHOLD, the validator becomes a valid candidate to propose a block for that slot. Otherwise, the validator skips the slot.
+
+As a result, there may be multiple validators eligible to propose a block for a slot. In this case, the block accepted by other nodes will prevail, provided it is on the chain with the latest finalized block as determined by the GRANDPA finality gadget. It's also possible for no block producers to be available for a slot, in which case the AURA consensus takes over. AURA is a fallback mechanism that randomly selects a validator to produce a block, running in parallel with BABE and only stepping in when no block producers exist for a slot. Otherwise, it remains inactive.
+
+Because validators roll independently, no block candidates may appear in some slots if all roll numbers are above the threshold.
+
```

Comment:
@0xLucca wdyt about adding here this reference: https://paritytech.github.io/polkadot-sdk/master/sp_consensus_babe/fn.make_vrf_sign_data.html

Something like: "To see an example of the implementation of the VRF in the polkadot-sdk, you can check out..."